b2sdk/sync/sync.py

```diff
@@ -189,37 +195,24 @@ def sync_folders(self, source_folder, dest_folder, now_millis, reporter):
         queue_limit = self.max_workers + 1000
         sync_executor = BoundedQueueExecutor(unbounded_executor, queue_limit=queue_limit)
 
-        # First, start the thread that counts the local files. That's the operation
-        # that should be fastest, and it provides scale for the progress reporting.
-        local_folder = None
-        if source_folder.folder_type() == 'local':
-            local_folder = source_folder
-        if dest_folder.folder_type() == 'local':
-            local_folder = dest_folder
-        if local_folder is None:
-            raise ValueError('neither folder is a local folder')
-        if reporter:
-            sync_executor.submit(count_files, local_folder, reporter)
+        if source_type == 'local' and reporter is not None:
+            # Start the thread that counts the local files. That's the operation
+            # that should be fastest, and it provides scale for the progress reporting.
+            sync_executor.submit(count_files, source_folder, reporter, self.policies_manager)
 
         # Schedule each of the actions
         bucket = None
-        if source_folder.folder_type() == 'b2':
-            bucket = source_folder.bucket
-        if dest_folder.folder_type() == 'b2':
+        if dest_type == 'b2':
```

Comment:
maybe small comment that:
for b2-to-b2 files would be created (copied to) on destination folder only so the order in this `if` is meaningful :)

Reply:
Yes, but actually all the API actions should be done on the source bucket so it's ok

Reply:
copy is done _on the destination_, not on the source

Reply:
Ok